integration/dockerfiles/Dockerfile_test_add

```diff
@@ -24,4 +24,7 @@ COPY $file /arg
 
 # Finally, test adding a remote URL, concurrently with a normal file
 ADD https://github.com/GoogleCloudPlatform/docker-credential-gcr/releases/download/v1.4.3/docker-credential-gcr_linux_386-1.4.3.tar.gz context/foo /test/all/
-ADD https://github.com/GoogleCloudPlatform/docker-credential-gcr/releases/download/v1.4.3-static/docker-credential-gcr_linux_amd64-1.4.3.tar.gz /destination
+
+# Test environment replacement in the URL
+ENV VERSION=v1.4.3
+ADD https://github.com/GoogleCloudPlatform/docker-credential-gcr/releases/download/${VERSION}-static/docker-credential-gcr_linux_amd64-1.4.3.tar.gz /destination
```

Comment:
Should this also do something to ensure the file was fetched and named correctly?

`RUN ls /destination/docker-credential-gcr_linux_amd64-1.4.3.tar.gz`

Or even better, `tar xcvf` it to make sure you actually got a `.tar.gz` and not a 404 response.

Reply:
I could add that here, but we actually get that for free. The way these tests work is that we build each of these with Docker and Kaniko, then compare the two images.